src/chk/chk_engine.c

```diff
@@ -782,13 +865,17 @@ chk_engine_start(uint64_t gen, uint32_t rank_nr, d_rank_t *ranks,
 out_pool:
 	d_list_for_each_entry_safe(cpr, tmp, &ins->ci_pool_list, cpr_link)
 		chk_pool_stop_one(ins, cpr->cpr_uuid,
-				  CHK__CHECK_POOL_STATUS__CPS_IMPLICATED, false);
+				  CHK__CHECK_POOL_STATUS__CPS_IMPLICATED, false, false);
+out_tree:
+	chk_destroy_pending_tree(ins);
+	chk_destroy_pool_tree(ins);
 out_iv:
-	ds_iv_ns_put(ins->ci_iv_ns);
-	ins->ci_iv_ns = NULL;
+	chk_iv_ns_cleanup(&ins->ci_iv_ns);
```

Comment:
Only for non-leader case? or for both cases?

Reply:
For both.